### DISCUSSION TITLE
TAPO TC85 - Failed to connect / load stream

### DISCUSSION THREAD

**Discussion by NoHairLeft4HKSV** [koush/scrypted#1378] [General]
Hello,

I am having an issue with a new set of Tapo TC85's that I just purchased. I have Scrypted setup and running on Unraid and already have a Tapo C310 working using the ONVIF plugin and hooked into my HKSV. I followed the same steps I used for the C310, adding the camera via the Tapo app, updating the firmware, setting the RTSP username and password (No special characters), On 2.4ghz connection, added to Scrypted, added the authentication and IP address (port 2020) and still no dice with getting the cameras to work. The cameras work fine on my home network with the Tapo app but refuse to work with Scrypted. 

Thinking I may setup something wrong or selected a setting incorrectly in the camera and or Scrypted, I factory erased the TC85's, deleted the Unraid docker for Scrypted (Purged the files from both cache and array), reinstalled Scrypted and setup my C310 again and then tried to add in the TC85. The C310 is working great but the TC85 refuses to produce a snapshot or a live feed. I have even tried the RTSP connection via VLC and can't seem to make a connection. Stream1 and Stream2 both fail. I installed the RTSP plugin in Scrypted and tried setting up a connection to the camera but also failed to connect / display anything.

Any advice or helpful pointers would be appreciated!

TC85 Firmware - 1.3.4 Build 231222 Rel. 399944n
C310 Firmware - 1.3.11 Build 231116 Rel. 69367n(5553)

Here is the console logs for one of the cameras. I've only tried adding 2 of the 4 back after resetting everything but both of the outputs look the same. Both cameras have different IP's, on the same subnet, found on my lan, authentication is enabled (No special characters), ETC.

linux x64 #1 SMP PREEMPT_DYNAMIC Wed Jun 28 07:51:54 PDT 2023
server version: 0.93.0
plugin version: @scrypted/onvif 0.1.10
full
[Rebroadcast Plugin]: error retrieving getVideoStreamOptions RPCResultError: Operation Timed Out
[Rebroadcast Plugin]:     at Timeout._onTimeout (/common/src/promise-utils.ts:32:43)
[Rebroadcast Plugin]:     at listOnTimeout (node:internal/timers:569:17)
[Rebroadcast Plugin]:     at processTimers (node:internal/timers:512:7)
[Rebroadcast Plugin]: @scrypted/onvif:host
[Rebroadcast Plugin]: host:@scrypted/prebuffer-mixin
[Rebroadcast Plugin]: host:@scrypted/prebuffer-mixin {
[Rebroadcast Plugin]:   cause: undefined
[Rebroadcast Plugin]: }
[Rebroadcast Plugin]: error retrieving getVideoStreamOptions RPCResultError: Operation Timed Out
[Rebroadcast Plugin]:     at Timeout._onTimeout (/common/src/promise-utils.ts:32:43)
[Rebroadcast Plugin]:     at listOnTimeout (node:internal/timers:569:17)
[Rebroadcast Plugin]:     at processTimers (node:internal/timers:512:7)
[Rebroadcast Plugin]: @scrypted/onvif:host
[Rebroadcast Plugin]: host:@scrypted/prebuffer-mixin
[Rebroadcast Plugin]: host:@scrypted/prebuffer-mixin {
[Rebroadcast Plugin]:   cause: undefined
[Rebroadcast Plugin]: }
error retrieving onvif profiles Error: Network timeout
    at ClientRequest.<anonymous> (/onvif/lib/cam.js:312:12)
    at Object.onceWrapper (node:events:631:28)
    at ClientRequest.emit (node:events:517:28)
    at ClientRequest.emit (node:domain:489:12)
    at Socket.emitRequestTimeout (node:_http_client:847:9)
    at Object.onceWrapper (node:events:631:28)
    at Socket.emit (node:events:529:35)
    at Socket.emit (node:domain:489:12)
    at Socket._onTimeout (node:net:598:8)
    at listOnTimeout (node:internal/timers:569:17)
listen loop connection failed, restarting listener. Network timeout
########################
3/13/2024, 5:45:26 PM
########################
error retrieving onvif profiles Error: Network timeout
    at ClientRequest.<anonymous> (/onvif/lib/cam.js:312:12)
    at Object.onceWrapper (node:events:631:28)
    at ClientRequest.emit (node:events:517:28)
    at ClientRequest.emit (node:domain:489:12)
    at Socket.emitRequestTimeout (node:_http_client:847:9)
    at Object.onceWrapper (node:events:631:28)
    at Socket.emit (node:events:529:35)
    at Socket.emit (node:domain:489:12)
    at Socket._onTimeout (node:net:598:8)
    at listOnTimeout (node:internal/timers:569:17)
listen loop connection failed, restarting listener. Network timeout
listen loop connection failed, restarting listener. Network timeout
########################
3/13/2024, 5:51:17 PM
########################
listen loop connection failed, restarting listener. Network timeout
[Rebroadcast Plugin]: error retrieving getVideoStreamOptions RPCResultError: Operation Timed Out
[Rebroadcast Plugin]:     at Timeout._onTimeout (/common/src/promise-utils.ts:32:43)
[Rebroadcast Plugin]:     at listOnTimeout (node:internal/timers:569:17)
[Rebroadcast Plugin]:     at processTimers (node:internal/timers:512:7)
[Rebroadcast Plugin]: @scrypted/onvif:host
[Rebroadcast Plugin]: host:@scrypted/prebuffer-mixin
[Rebroadcast Plugin]: host:@scrypted/prebuffer-mixin {
[Rebroadcast Plugin]:   cause: undefined
[Rebroadcast Plugin]: }
listen loop connection failed, restarting listener. Network timeout
error retrieving onvif profiles Error: Network timeout
    at ClientRequest.<anonymous> (/onvif/lib/cam.js:312:12)
    at Object.onceWrapper (node:events:631:28)
    at ClientRequest.emit (node:events:517:28)
    at ClientRequest.emit (node:domain:489:12)
    at Socket.emitRequestTimeout (node:_http_client:847:9)
    at Object.onceWrapper (node:events:631:28)
    at Socket.emit (node:events:529:35)
    at Socket.emit (node:domain:489:12)
    at Socket._onTimeout (node:net:598:8)
    at listOnTimeout (node:internal/timers:569:17)
[Snapshot Plugin]: creating error image with background Snapshot Failed false
listen loop connection failed, restarting listener. Network timeout
error retrieving onvif profiles Error: Network timeout
    at ClientRequest.<anonymous> (/onvif/lib/cam.js:312:12)
    at Object.onceWrapper (node:events:631:28)
    at ClientRequest.emit (node:events:517:28)
    at ClientRequest.emit (node:domain:489:12)
    at Socket.emitRequestTimeout (node:_http_client:847:9)
    at Object.onceWrapper (node:events:631:28)
    at Socket.emit (node:events:529:35)
    at Socket.emit (node:domain:489:12)
    at Socket._onTimeout (node:net:598:8)
    at listOnTimeout (node:internal/timers:569:17)
listen loop connection failed, restarting listener. Network timeout
listen loop connection failed, restarting listener. Network timeout
listen loop connection failed, restarting listener. Network timeout
listen loop connection failed, restarting listener. Network timeout

**Comment by NoHairLeft4HKSV** [comment]
I tried again this morning to make a connection and ping the cameras, no luck getting any 3rd party access and ping requests time out. I then left a terminal running pinging one of the cameras and deleted the RTSP setup and when I hit save, I started getting replies on the terminal. They quickly stopped so I went into the live feed via the Tapo app and the ping replies came flooding in. I then tried adding the RTSP account back to the camera, left the ping terminal running and went to live feed in hopes of getting scrypted to see the device. Below is a capture from the console.


[WebRTC Plugin]: signalingState have-remote-offer
[WebRTC Plugin]: iceGatheringState gathering
[WebRTC Plugin]: signalingState stable
[WebRTC Plugin]: connectionState connecting
[WebRTC Plugin]: iceConnectionState checking
[WebRTC Plugin]: waiting ice connected
[WebRTC Plugin]: iceConnectionState completed
[WebRTC Plugin]: iceGatheringState complete
[WebRTC Plugin]: iceConnectionState connected
[WebRTC Plugin]: connectionState connected
[WebRTC Plugin]: Connection is local network: true 192.168.1.138 { ipv4: true, isLocalNetwork: true, destinationId: '192.168.1.138' }
[WebRTC Plugin]: done waiting ice connected
[WebRTC Plugin]: Connection is local network: true 192.168.1.138 { ipv4: true, isLocalNetwork: true, destinationId: '192.168.1.138' }
error retrieving onvif profiles Error: connect ECONNREFUSED 192.168.1.122:2020
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1555:16)
    at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:128:17) {
  errno: -111,
  code: 'ECONNREFUSED',
  syscall: 'connect',
  address: '192.168.1.122',
  port: 2020
}
[WebRTC Plugin]: Error starting playback for WebRTC track. RPCResultError [TypeError]: Cannot read properties of undefined (reading 'allowBatteryPrebuffer')
[WebRTC Plugin]:     at G.ensurePrebufferSessions (/@scrypted/prebuffer-mixin/main.nodejs.js:1:129045)
[WebRTC Plugin]:     at async G.getVideoStream (/@scrypted/prebuffer-mixin/main.nodejs.js:1:125922)
[WebRTC Plugin]:     at async RpcPeer.handleMessageInternal (/server/node_modules/@scrypted/server/dist/rpc.js:601:37)
[WebRTC Plugin]: thread:main
[WebRTC Plugin]: @scrypted/prebuffer-mixin:host
[WebRTC Plugin]: host:@scrypted/webrtc {
[WebRTC Plugin]:   cause: undefined
[WebRTC Plugin]: }
[WebRTC Plugin]: signalingState closed
[WebRTC Plugin]: connectionState closed
[WebRTC Plugin]: iceConnectionState closed
listen loop connection failed, restarting listener. connect ECONNREFUSED 192.168.1.122:2020
listen loop connection failed, restarting listener. connect ECONNREFUSED 192.168.1.122:2020
listen loop connection failed, restarting listener. connect ECONNREFUSED 192.168.1.122:2020
listen loop connection failed, restarting listener. connect ECONNREFUSED 192.168.1.122:2020
listen loop connection failed, restarting listener. connect ECONNREFUSED 192.168.1.122:2020
listen loop connection failed, restarting listener. connect ECONNREFUSED 192.168.1.122:2020
listen loop connection failed, restarting listener. connect ECONNREFUSED 192.168.1.122:2020
listen loop connection failed, restarting listener. connect ECONNREFUSED 192.168.1.122:2020
listen loop connection failed, restarting listener. connect ECONNREFUSED 192.168.1.122:2020
listen loop connection failed, restarting listener. connect ECONNREFUSED 192.168.1.122:2020
listen loop connection failed, restarting listener. connect ECONNREFUSED 192.168.1.122:2020
listen loop connection failed, restarting listener. connect ECONNREFUSED 192.168.1.122:2020


Looks like the cameras are not staying active even when valid setup is implemented. I went ahead and submitted a Tapo support request asking if they had a different firmware to support this feature.

**Comment by cubbieblue16** [comment]
I get this rebroadcast issue with my Google Cameras as well.

**Comment by NoHairLeft4HKSV** [comment]
Update - Per TAPO the TC85's do not include firmware to allow for 3rd party access even though the app indicates the options. I did politely ask that TP / Tapo work on this as the cameras would be a hot seller if they could enable the integration feature. As of right now I just popped in Micro SD cards and I'm using the TAPO app to view the streams. Also I bought cheap solar panels off Amazon and connected them to the cameras outside and they are working and charging without issue. The speed over my Wi-Fi network is very quick and I have had no drops or other issues.

Maybe in the future Tapo will release a firmware to support 3rd party access but as of 03.29.2024 we are out of luck.
